core/src/main/java/com/megaease/easeagent/core/Register.java

```diff
@@ -26,25 +26,34 @@
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Method;
+import java.net.URLClassLoader;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.WeakHashMap;
 
 import static com.google.common.collect.FluentIterable.from;
 
 class Register {
     private static final Logger LOGGER = LoggerFactory.getLogger(Register.class);
 
     private final Iterable<QualifiedBean> beans;
-    private final Set<Integer> applied;
+    private final WeakHashMap<ClassLoader, Set<String>> applied = new WeakHashMap<>();
 
     Register(Iterable<QualifiedBean> beans) {
         this.beans = beans;
-        applied = new HashSet<Integer>();
     }
 
-    void apply(String adviceClassName, ClassLoader external) {
-        if (!applied.add(Objects.hashCode(external, adviceClassName))) return;
+    synchronized void apply(String adviceClassName, ClassLoader external) {
```

Comment:
The instance of `adviceClassName` should be created once.
If adviceClassName = "a.b.UserAdvice":

1. `external = a`, Set<String> classNames = applied.get(external); ClassNames will be null, and classNames.add(adviceClassName) return true, then nwInstanceOf will be invoked.

2. `external = b`, Set<String> classNames = applied.get(external); ClassNames will be null, and classNames.add(adviceClassName) return true, then newInstanceOf will be invoked.

The instance of `adviceClassName` be created twice.